tests/cli/functional/conftest.py

```diff
@@ -0,0 +1,112 @@
+"""
+This file is part of nucypher.
+
+nucypher is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+nucypher is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+import os
+import pytest
+from eth_account import Account
+
+from nucypher.blockchain.economics import EconomicsFactory
+from nucypher.blockchain.eth.agents import ContractAgency
+from nucypher.blockchain.eth.interfaces import BlockchainInterface, BlockchainInterfaceFactory
+from nucypher.blockchain.eth.registry import InMemoryContractRegistry
+from nucypher.config.characters import UrsulaConfiguration
+from nucypher.utilities.sandbox.constants import MOCK_PROVIDER_URI
+from tests.cli.functional.test_ursula_local_keystore_cli_functionality import NUMBER_OF_MOCK_ACCOUNTS, \
+    KEYFILE_NAME_TEMPLATE
+from tests.fixtures import _make_testerchain, make_token_economics
+from tests.mock.interfaces import make_mock_registry_source_manager, MockBlockchain
+from tests.mock.agents import FAKE_RECEIPT, MockContractAgency
+
+
+@pytest.fixture(scope='module', autouse=True)
+def mock_testerchain() -> MockBlockchain:
+    BlockchainInterfaceFactory._interfaces = dict()
+    testerchain = _make_testerchain(mock_backend=True)
+    BlockchainInterfaceFactory.register_interface(interface=testerchain)
+    yield testerchain
+
+
+@pytest.fixture(scope='module')
+def token_economics(mock_testerchain):
+    return make_token_economics(blockchain=mock_testerchain)
+
+
+@pytest.fixture(scope='module', autouse=True)
+def mock_interface(module_mocker):
+    mock_transaction_sender = module_mocker.patch.object(BlockchainInterface, 'sign_and_broadcast_transaction')
+    mock_transaction_sender.return_value = FAKE_RECEIPT
+    return mock_transaction_sender
+
+
+@pytest.fixture(scope='session')
```

Comment:
Why is this fixture session-scoped? I think it's riskier and I don't immediately see the benefit. If for some reason some test in a test module alters the registry, there will be side-effects in other test modules. Debugging that can be crazy. 

```suggestion
@pytest.fixture(scope='module')
```

Reply:
In principle, the contract registry will never be hit, instead mocked at the agent layer, however, it's been easier thus far to let some objects handle the empty registry anyways.  

Since we aspire to make this fixture unreachable, I'm not sure what makes the most sense in the interim...  In favor of your suggestion, we might as well set it to functional scoped to ensure each test enjoys a little more isolation.

what do you think?

Reply:
:heavy_check_mark: